models/clip/extract_clip.py

```diff
@@ -0,0 +1,175 @@
+# import traceback
+
+
+class ExtractCLIP(torch.nn.Module):
+
+    def __init__(self, args):
+        super(ExtractCLIP, self).__init__()
+        self.feature_type = args.feature_type
+        self.path_list = form_list_from_user_input(args)
+        self.batch_size = args.batch_size
+        self.extraction_fps = args.extraction_fps
+        # not used, create an issue if you would like to save the frames
+        self.keep_tmp_files = args.keep_tmp_files
+        self.on_extraction = args.on_extraction
+        self.tmp_path = os.path.join(args.tmp_path, self.feature_type)
+        self.output_path = os.path.join(args.output_path, self.feature_type)
+        self.progress = tqdm(total=len(self.path_list))
+
+    def forward(self, indices: torch.LongTensor):
+        '''
+        Arguments:
+            indices {torch.LongTensor} -- indices to self.path_list
+        '''
+        device = indices.device
+        model, preprocess = self.load_model(device)
+
+        for idx in indices:
+            # when error occurs might fail silently when run from torch data parallel
+            try:
+                feats_dict = self.extract(device, model, preprocess, self.path_list[idx])
+                action_on_extraction(feats_dict, self.path_list[idx], self.output_path, self.on_extraction)
+            except KeyboardInterrupt:
+                raise KeyboardInterrupt
+            except Exception as e:
+                # prints only the last line of an error. Use `traceback.print_exc()` for the whole traceback
+                print(e)
+                print(f'Extraction failed at: {self.path_list[idx]} with error (↑). Continuing extraction')
+
+            # update tqdm progress bar
+            self.progress.update()
+
+    def extract(self, device: torch.device, model: torch.nn.Module, preprocess: Callable,
+                video_path: Union[str, None] = None) -> Dict[str, np.ndarray]:
+        '''The extraction call. Made to clean the forward call a bit.
+
+        Arguments:
+            device {torch.device}
+            model {torch.nn.Module}
+            preprocess {Callable} -- function to preprocess images
+
+        Keyword Arguments:
+            video_path {Union[str, None]} -- if you would like to use import it and use it as
+                                             "path -> model"-fashion (default: {None})
+
+        Returns:
+            Dict[str, np.ndarray]: 'features_nme', 'fps', 'timestamps_ms'
+        '''
+        def _run_on_a_batch(vid_feats, batch, model, device):
+            batch = torch.cat(batch).to(device)
+
+            with torch.no_grad():
+                batch_feats = model.encode_image(batch)
+                vid_feats.extend(batch_feats.tolist())
+                # show predicitons on imagenet dataset (might be useful for debugging)
+                # if self.show_pred:
+                #     logits = classifier(batch_feats)
```

Comment:
we need to find a way :)

Reply:
I think we can ask the user to provide a sentence and then calculate the similarity of the video to it. Just like the first [CLIP example](https://github.com/openai/CLIP/blob/main/README.md#usage), since this task should be more in line with the training method. This would import an extra configuration item, but it shouldn't be a big problem.
Of course we could also use Zero-Shot Prediction as you mentioned, since apparently they already offer this as a demonstration.